src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,286 @@
+
+    # Unsigned right bitshifting
+    def bitshiftrightzero(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        if (op1 >= 0): # For positive operand 1 it works the same as bitshiftright
+            sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        else: # For negative operand 1 it needs conversion to unsigned
+            sum = (unsigned32(op1) >> (op2 % (_constants.bitmax_exposant + 1)))
+            if (sum > _constants.bitmax):
+                op2 = 1  # This way it guarantees that the summa does not go above maximum
+                sum = (unsigned32(op1) >> (op2 % (_constants.bitmax_exposant + 1)))
+        return {'oper': '>>>', 'op2': op2, 'sum': sum}
+
+    # Bitwise and
+    def bitand(self, op1, op2):
+        sum = op1 & op2
+        return {'oper': '&', 'op2': op2, 'sum': sum}
+
+    # Bitwise or
+    def bitor(self, op1, op2):
+        sum = op1 | op2
+        return {'oper': '|', 'op2': op2, 'sum': sum}
+
+    # Bitwise xor (exclusive or)
+    def bitxor(self, op1, op2):
+        sum = op2 ^ op1
+        return {'oper': '^', 'op2': op2, 'sum': sum}
+
+# Generator function
+def generate_binary_tests(options):
+    if not os.path.isdir(options.output):
+        os.makedirs(options.output)
+    SEED = options.seed
+    random.seed(SEED)
+    generated_test_counter = 0 # How many tests are written (defending against script error: null)
+    file_count = 1 # How many files are created / written
+    new_file_begin = True # If new file is opened / created
+    bitcalc = False # If we need to do bitwise calculations
+    arithcalc = False # If we need to do arithmetic calculations
+    bit = False # Actually are we doing bitwise or arithmetic calculations
+    for op in options.op:
+        # If arithmetic operators are currently needed
+        if ((op == "add") or (op == "sub") or (op == "mul") or (op == "div") or
+        (op == "mod")):
+            arithcalc = True
+
+        # If bitwise operators are currently needed
+        elif ((op == "xor") or (op == "and") or (op == "or") or (op == "shiftright")
+        or (op == "shiftleft") or (op == "shiftrightzero")):
+            bitcalc = True
+
+    file_to_open = os.path.join(options.output, _GENERATED_FILENAME.format(NUMBER = file_count))
+    for i in range(options.test_count):
+        generated_test_counter += 1
+        if (generated_test_counter % _constants.test_case_in_a_file == 0): # If a new file is needed
+            new_file_begin = True
+            file_count += 1
+            file_to_open = os.path.join(options.output, _GENERATED_FILENAME.format(NUMBER = file_count))
+        if (new_file_begin and os.path.isfile(file_to_open)): # If file already exists this deletes it
+            os.remove(file_to_open)
+        with open(file_to_open, "a") as file: # If new file begins validate function needs to be added
+            if (new_file_begin):
+                print(file_to_open)
```

Comment:
Is there a need for this printing?

Reply:
@repasics also included it in his script, and this is the only way to know which files we generated. (If we generate 5 tests and run it, and its correct, and we still get an error, it can be because of a previous generation of a higher test count) and this way we know it it could be because of that. And if we are generating like 50.000+ tests, it is good to see where we are in the generating. And the test runner script does not shows us this. I think we need it here and it is useful.
https://github.com/szeged/js-test-generators/blob/69e7b7f2faffd704d139bac1bcf509dfc24d2c29/src/scripts/generate-unary-ops.py#L39